compiler/rustc_middle/src/mir/visit.rs

```diff
@@ -1114,6 +1114,7 @@ macro_rules! visit_place_fns {
                     if ty != new_ty { Some(PlaceElem::OpaqueCast(new_ty)) } else { None }
                 }
                 PlaceElem::Deref
+                | PlaceElem::Subtype { .. }
```

Comment:
this also needs to visit the `ty`, similar to `OpaqueCast`.

Reply:
> then this is a bug with the lowering. Subtype projections are still ordinary moves.

How ? We didn't even reach lowering yet this errs on `elaborate_drops`